kep/9-capacity-scheduling-for-ml-dl-workloads-based-on-scheduler-framework/README.md

```diff
@@ -0,0 +1,208 @@
+- min: it means the guaranteed resource of Quota.
+
+They act on different phases of pod lifecycle. So it maybe possible to merge them together.
+
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+We assume two elastic quotas are defined: QuotaA (min:4, max:6) and QuotaB (min:6, max:8), the quota unit is the number of GPUs. UserA in green color consumes QuotaA and UserB in red color consumes QuotaB. The entire cluster has 10 GPUs available hence the sum of quota min is equal to the cluster capacity.
+
+Initially, UserA consumes 2 GPUs and UserB consumes 3 GPUs. Both consumptions are within the reserved "min" quota range.
+![1](./1.png)
+
+
+Next, UserA consumes 4 GPUs reaching the "min" quota of QuotaA. UserB still consumes 3 GPUs. The cluster still has sufficient GPU resources hence no resource stealing happens.
+![2](./2.png)
+
+
+Assuming UserA attempts to use the rest of 3 unused GPUs in the cluster,  he can only use 2 more GPUs since his quota "max" is 6 GPUs. Note that these 2 GPUs are "borrowed" from UserB's "Guaranteed" quota since UserB does not use up it. 
+![3](./3.png)
+
+Later, if UserB demands 8 GPUs, 5 more compared to his current consumption, his request cannot be immediately satisfied since only one unused GPU left in the cluster. Since UserB's "min" quota is 6 GPUs, the preemption algorithm ensures that three GPUs will be returned from UserA.  However, no more GPUs can be assigned to UserB since UserA's "min" quota has to be honored as well. In the end, UserA consumes 4 GPUs and UserB consumes 6 GPUs, satisfying each user's "min" quota.
+![4](./4.png)
+
+## Design Details
+
+### Extention point
+
+#### PreFilter
+We will check if the (Pod.request + Quota.Allocated) is less than Quota.Max. If not, the scheduling cycle of PodA will fail. 
+
+#### PostFilter
+We will reimplement the method `FindCandidates` in `defaultPreempt`. The original `FindCandidates` method selects all the pods with the lower priority than the preemptor’s priority as potential victims in a node. 
+In this proposal, the victim selection logic needs to be changed in the following cases.
+- if Preemptor.Request + Quota.allocated > Quota.min: It means that its min or guaranteed resource is used or `borrowed` by other Quota. So we will ask the quotas who's allocated is more than max (it means thest quotas borrow the resource from other quota)
+
+- if Preemptor.Requst + Quota.allocated <= Quota.min: It means that its guaranteed isn't borrowed by other quotas. So that we will select the pods belongs to the same quota(namespace) with the lower priority than the preemptor’s priority as potential victims in a node.
```

Comment:
Requst -> Request

Reply:
Same question, should it be "_if Preemptor.Requst + Quota.allocated **>=** Quota.min_"?

Reply:
Thanks Done